  <tr>
        <td>
            <div align="left">
                <font size=25px>
                    <b> Predicting whether the person is Covid Positive or Negative
                    </b>
                </font>
            </div>
        </td>
    </tr>

## Importing Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Loading the Data

In [2]:
df = pd.read_excel(r'C:\Users\HELLO\Downloads\Covid_Dataset_new.xlsx',header=0)

In [3]:
df

,cough,fever,sore_throat,shortness_of_breath,head_ache,age_60_and_above,gender,test_indication,corona_result
0,1,1,0,0,0,Yes,male,Other,negative
1,0,0,0,0,0,No,male,Other,positive
2,0,0,0,0,0,Yes,female,Other,negative
3,1,0,0,0,0,No,male,Abroad,negative
4,0,0,0,0,0,Yes,male,Other,positive
...,...,...,...,...,...,...,...,...,...
2132,1,1,0,0,0,No,male,Contact with confirmed,positive
2133,1,0,0,0,1,No,female,Contact with confirmed,positive
2134,0,0,0,0,0,Yes,male,Other,negative
2135,0,0,0,0,0,No,male,Other,negative


In [4]:
df.shape

(2137, 9)

## Understanding the Data

In [5]:
df.describe(include="all")

,cough,fever,sore_throat,shortness_of_breath,head_ache,age_60_and_above,gender,test_indication,corona_result
count,2137.000000,2137.000000,2137.000000,2137.000000,2137.000000,2137,2137,2137,2137
unique,NaN,NaN,NaN,NaN,NaN,2,3,3,2
top,NaN,NaN,NaN,NaN,NaN,No,female,Other,positive
freq,NaN,NaN,NaN,NaN,NaN,1159,1064,1290,1105
mean,0.419747,0.273280,0.038372,0.037436,0.058493,NaN,NaN,NaN,NaN
std,0.493633,0.445748,0.192137,0.189871,0.234729,NaN,NaN,NaN,NaN
min,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,NaN,NaN,NaN
25%,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,NaN,NaN,NaN
50%,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,NaN,NaN,NaN
75%,1.000000,1.000000,0.000000,0.000000,0.000000,NaN,NaN,NaN,NaN


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2137 entries, 0 to 2136
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   cough                2137 non-null   int64 
 1   fever                2137 non-null   int64 
 2   sore_throat          2137 non-null   int64 
 3   shortness_of_breath  2137 non-null   int64 
 4   head_ache            2137 non-null   int64 
 5   age_60_and_above     2137 non-null   object
 6   gender               2137 non-null   object
 7   test_indication      2137 non-null   object
 8   corona_result        2137 non-null   object
dtypes: int64(5), object(4)
memory usage: 150.4+ KB


## Checking for Duplicated and Missing Values

In [7]:
df.isnull().sum()

cough                  0
fever                  0
sore_throat            0
shortness_of_breath    0
head_ache              0
age_60_and_above       0
gender                 0
test_indication        0
corona_result          0
dtype: int64

In [8]:
df.duplicated().sum()
# dont delete duplicates as it is large amount of data and there is possibility of the data to be same.

1925

In [9]:
colname=[]
for x in df.columns:
    if df[x].dtype=='object':
        colname.append(x)
colname

['age_60_and_above', 'gender', 'test_indication', 'corona_result']

## Converting Categorical to Numerical

In [10]:
from sklearn.preprocessing import LabelEncoder
 
le=LabelEncoder()
 
for x in colname:
    df[x]=le.fit_transform(df[x])


In [11]:
print(df.dtypes)
df.head()

cough                  int64
fever                  int64
sore_throat            int64
shortness_of_breath    int64
head_ache              int64
age_60_and_above       int32
gender                 int32
test_indication        int32
corona_result          int32
dtype: object


,cough,fever,sore_throat,shortness_of_breath,head_ache,age_60_and_above,gender,test_indication,corona_result
0,1,1,0,0,0,1,2,2,0
1,0,0,0,0,0,0,2,2,1
2,0,0,0,0,0,1,1,2,0
3,1,0,0,0,0,0,2,0,0
4,0,0,0,0,0,1,2,2,1


In [12]:
X = df.values[:,0:-1]
Y = df.values[:,-1]

## Standardization

In [13]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()

scaler.fit(X)
X = scaler.transform(X)

In [14]:
Y=Y.astype(int)

## Splitting the Data

In [15]:
from sklearn.model_selection import train_test_split
 
#Split the data into test and train
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, 
                                                    random_state=10)

## MODELS

In [19]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
 
# first, initialize the classificators
tree= DecisionTreeClassifier(random_state=10) # using the random state for reproducibility
knn= KNeighborsClassifier(n_neighbors=5,metric='euclidean')
svm= SVC(kernel="sigmoid", gamma=0.1, C=90,random_state=10)
logreg=LogisticRegression(multi_class="multinomial",random_state=10)

In [20]:
models= [tree, knn, svm, logreg]

### Accuracy for DT, SVC, KNN, Logistic regression

In [21]:
from sklearn.metrics import confusion_matrix, accuracy_score,classification_report
 
for model in models:
    model.fit(X_train, Y_train) # fit the model
    Y_pred= model.predict(X_test) # then predict on the test set
    accuracy= accuracy_score(Y_test, Y_pred) 
    clf_report= classification_report(Y_test, Y_pred) 
    print(confusion_matrix(Y_test,Y_pred))
    print("The accuracy of the ",type(model).__name__, " model is ", accuracy*100 )
    print("Classification report:\n", clf_report)
    print("\n")

[[167  36]
 [ 63 162]]
The accuracy of the  DecisionTreeClassifier  model is  76.86915887850468
Classification report:
               precision    recall  f1-score   support

           0       0.73      0.82      0.77       203
           1       0.82      0.72      0.77       225

    accuracy                           0.77       428
   macro avg       0.77      0.77      0.77       428
weighted avg       0.77      0.77      0.77       428



[[147  56]
 [ 53 172]]
The accuracy of the  KNeighborsClassifier  model is  74.53271028037383
Classification report:
               precision    recall  f1-score   support

           0       0.73      0.72      0.73       203
           1       0.75      0.76      0.76       225

    accuracy                           0.75       428
   macro avg       0.74      0.74      0.74       428
weighted avg       0.75      0.75      0.75       428



[[115  88]
 [ 88 137]]
The accuracy of the  SVC  model is  58.87850467289719
Classification report:
    

## Random Forest 

In [22]:
from sklearn.ensemble import RandomForestClassifier
 
model_RandomForest=RandomForestClassifier(n_estimators=500,#500
                                          random_state=10, bootstrap=True,
                                         n_jobs=-1)
 
#fit the model on the data and predict the values
model_RandomForest.fit(X_train,Y_train)
 
Y_pred=model_RandomForest.predict(X_test)

#### Random forest evaluation

In [23]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

cfm=confusion_matrix(Y_test,Y_pred)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,Y_pred))
 
acc=accuracy_score(Y_test, Y_pred)
print("Accuracy of the model: ",acc)

[[167  36]
 [ 62 163]]
Classification report: 
              precision    recall  f1-score   support

           0       0.73      0.82      0.77       203
           1       0.82      0.72      0.77       225

    accuracy                           0.77       428
   macro avg       0.77      0.77      0.77       428
weighted avg       0.78      0.77      0.77       428

Accuracy of the model:  0.7710280373831776


## Extra Trees

In [24]:
from sklearn.ensemble import ExtraTreesClassifier
 
model_EXT=ExtraTreesClassifier(n_estimators=100,# how many trees to grow
                                          random_state=10, bootstrap=True,#bootstrap is false by default,in large data put bootstrap as false in extra trees
                                         n_jobs=-1)
 
#fit the model on the data and predict the values
model_EXT.fit(X_train,Y_train)
 
Y_pred=model_EXT.predict(X_test)

#### EXT  Evaluation

In [25]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

cfm=confusion_matrix(Y_test,Y_pred)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,Y_pred))
 
acc=accuracy_score(Y_test, Y_pred)
print("Accuracy of the model: ",acc)

[[167  36]
 [ 58 167]]
Classification report: 
              precision    recall  f1-score   support

           0       0.74      0.82      0.78       203
           1       0.82      0.74      0.78       225

    accuracy                           0.78       428
   macro avg       0.78      0.78      0.78       428
weighted avg       0.78      0.78      0.78       428

Accuracy of the model:  0.780373831775701


### Tuning the model

In [26]:
classifier=LogisticRegression()
# fitting training data to the model
classifier.fit(X_train,Y_train)

LogisticRegression()

In [27]:
y_pred_prob=classifier.predict_proba(X_test)
print(y_pred_prob)

[[6.05601615e-03 9.93943984e-01]
 [2.50692073e-01 7.49307927e-01]
 [4.80486251e-01 5.19513749e-01]
 [2.95494678e-01 7.04505322e-01]
 [2.98204079e-01 7.01795921e-01]
 [4.18131126e-01 5.81868874e-01]
 [6.13108099e-01 3.86891901e-01]
 [2.66574928e-01 7.33425072e-01]
 [4.56548749e-01 5.43451251e-01]
 [7.11556566e-01 2.88443434e-01]
 [4.80486251e-01 5.19513749e-01]
 [2.80935016e-03 9.97190650e-01]
 [2.98204079e-01 7.01795921e-01]
 [7.11556566e-01 2.88443434e-01]
 [5.36928859e-01 4.63071141e-01]
 [7.58049583e-01 2.41950417e-01]
 [7.11556566e-01 2.88443434e-01]
 [5.55109171e-01 4.44890829e-01]
 [4.56548749e-01 5.43451251e-01]
 [7.11556566e-01 2.88443434e-01]
 [4.80486251e-01 5.19513749e-01]
 [3.98120008e-01 6.01879992e-01]
 [7.11556566e-01 2.88443434e-01]
 [2.43561300e-03 9.97564387e-01]
 [4.56548749e-01 5.43451251e-01]
 [7.11556566e-01 2.88443434e-01]
 [6.46462444e-01 3.53537556e-01]
 [2.44297780e-05 9.99975570e-01]
 [5.55109171e-01 4.44890829e-01]
 [5.55109171e-01 4.44890829e-01]
 [5.551091

In [28]:
y_pred_class=[]
for value in y_pred_prob[:,1]:
    if value > 0.48:
        y_pred_class.append(1)
    else:
        y_pred_class.append(0)
print(y_pred_class)

[1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 

In [29]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report
 
cfm=confusion_matrix(Y_test,y_pred_class)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,y_pred_class))
 
acc=accuracy_score(Y_test, y_pred_class)
print("Accuracy of the model: ",acc)

[[135  68]
 [ 74 151]]
Classification report: 
              precision    recall  f1-score   support

           0       0.65      0.67      0.66       203
           1       0.69      0.67      0.68       225

    accuracy                           0.67       428
   macro avg       0.67      0.67      0.67       428
weighted avg       0.67      0.67      0.67       428

Accuracy of the model:  0.6682242990654206


In [30]:
for a in np.arange(0.4,0.61,0.01):
    predict_mine = np.where(y_pred_prob[:,1] > a, 1, 0)
    cfm=confusion_matrix(Y_test, predict_mine)
    total_err=cfm[0,1]+cfm[1,0]
    print("Errors at threshold ", a, ":",total_err, " , type 2 error :", 
          cfm[1,0]," , type 1 error:", cfm[0,1])

Errors at threshold  0.4 : 143  , type 2 error : 39  , type 1 error: 104
Errors at threshold  0.41000000000000003 : 143  , type 2 error : 39  , type 1 error: 104
Errors at threshold  0.42000000000000004 : 143  , type 2 error : 39  , type 1 error: 104
Errors at threshold  0.43000000000000005 : 149  , type 2 error : 46  , type 1 error: 103
Errors at threshold  0.44000000000000006 : 149  , type 2 error : 46  , type 1 error: 103
Errors at threshold  0.45000000000000007 : 156  , type 2 error : 60  , type 1 error: 96
Errors at threshold  0.4600000000000001 : 153  , type 2 error : 68  , type 1 error: 85
Errors at threshold  0.4700000000000001 : 142  , type 2 error : 74  , type 1 error: 68
Errors at threshold  0.4800000000000001 : 142  , type 2 error : 74  , type 1 error: 68
Errors at threshold  0.4900000000000001 : 149  , type 2 error : 81  , type 1 error: 68
Errors at threshold  0.5000000000000001 : 153  , type 2 error : 87  , type 1 error: 66
Errors at threshold  0.5100000000000001 : 156  ,

### Tuned Decision Tree

In [31]:
from sklearn.tree import DecisionTreeClassifier

model_DT=DecisionTreeClassifier(random_state=10,criterion='gini',splitter='best',min_samples_leaf=3,
                               min_samples_split=5,max_depth=10)

# min_samples_leaf, min_samples_split, max_depth, max_features, max_leaf_nodes[pruning]

# fit the model on the data and predict the values
model_DT.fit(X_train,Y_train)
Y_pred=model_DT.predict(X_test)
Y_pred

array([0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1,
       0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1,
       1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1,
       1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0,
       0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0,
       1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0,
       0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1,
       0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1,
       1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0,

In [32]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

cfm=confusion_matrix(Y_test,Y_pred)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,Y_pred))
 
acc=accuracy_score(Y_test, Y_pred)
print("Accuracy of the model: ",acc)

[[167  36]
 [ 66 159]]
Classification report: 
              precision    recall  f1-score   support

           0       0.72      0.82      0.77       203
           1       0.82      0.71      0.76       225

    accuracy                           0.76       428
   macro avg       0.77      0.76      0.76       428
weighted avg       0.77      0.76      0.76       428

Accuracy of the model:  0.7616822429906542


## Grid Search CV

In [39]:
from sklearn.ensemble import ExtraTreesClassifier
 
model_EXT=ExtraTreesClassifier( random_state=10, bootstrap=True)
 
parameter_space = {
    'n_estimators':[100,300,500,1000],       #np.arange(100, 1001,50),
    'max_depth':[10,15, 8, 12],
    'min_samples_leaf':[3,4,5,6,7]
    }
from sklearn.model_selection import GridSearchCV  #RandomizedSearchCV
clf = GridSearchCV(model_EXT, parameter_space, n_jobs=-1, cv=5)

In [42]:
clf.fit(X_train,Y_train)

GridSearchCV(cv=5,
             estimator=ExtraTreesClassifier(bootstrap=True, random_state=10),
             n_jobs=-1,
             param_grid={'max_depth': [10, 15, 8, 12],
                         'min_samples_leaf': [3, 4, 5, 6, 7],
                         'n_estimators': [100, 300, 500, 1000]})

In [43]:
print('Best parameters found:\n', clf.best_params_)
clf.best_score_ 

Best parameters found:
 {'max_depth': 10, 'min_samples_leaf': 3, 'n_estimators': 1000}


0.794021711169419

In [44]:
Y_pred=clf.predict(X_test)

In [45]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

cfm=confusion_matrix(Y_test,Y_pred)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,Y_pred))
 
acc=accuracy_score(Y_test, Y_pred)
print("Accuracy of the model: ",acc)

[[167  36]
 [ 58 167]]
Classification report: 
              precision    recall  f1-score   support

           0       0.74      0.82      0.78       203
           1       0.82      0.74      0.78       225

    accuracy                           0.78       428
   macro avg       0.78      0.78      0.78       428
weighted avg       0.78      0.78      0.78       428

Accuracy of the model:  0.780373831775701


In [46]:
from sklearn.tree import DecisionTreeClassifier

model_DT=DecisionTreeClassifier(random_state=10,criterion='gini',min_samples_leaf=7,
                               min_samples_split=5,max_depth=15)

# min_samples_leaf, min_samples_split, max_depth, max_features, max_leaf_nodes[pruning]

# fit the model on the data and predict the values
model_DT.fit(X_train,Y_train)
Y_pred=model_DT.predict(X_test)
Y_pred

array([0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1,
       0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1,
       1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1,
       1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0,
       0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0,
       1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0,
       0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1,
       0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0,
       1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0,

In [47]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

cfm=confusion_matrix(Y_test,Y_pred)
print(cfm)
 
print("Classification report: ")
 
print(classification_report(Y_test,Y_pred))
 
acc=accuracy_score(Y_test, Y_pred)
print("Accuracy of the model: ",acc)

[[167  36]
 [ 70 155]]
Classification report: 
              precision    recall  f1-score   support

           0       0.70      0.82      0.76       203
           1       0.81      0.69      0.75       225

    accuracy                           0.75       428
   macro avg       0.76      0.76      0.75       428
weighted avg       0.76      0.75      0.75       428

Accuracy of the model:  0.7523364485981309


In [48]:
model_DT.score(X_train,Y_train)

0.7922761849034523

### Conclusion :
* Extra Trees Classifier worked best on this data as it gave a good recall comparatively for class1 which is class of importance as well as the best accuracy of 0.78. 